### PR TITLE
Update supported operating systems for dualstack

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
Update the dualstack tests for newly integrated operating systems. 

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by PratikDeoghare** [comment]
/hold 

make tests optional before merging

File: hack/ci/testdata/kubermatic_dualstack.yaml
Comment (ahmedwaleedmalik) [inline hack/ci/testdata/kubermatic_dualstack.yaml]: We don't need to pin these images any more as https://github.com/kubermatic/kubermatic/pull/10856 will update KKP to use the latest available images

File: hack/ci/run-dualstack-e2e-test.sh
Comment (ahmedwaleedmalik) [inline hack/ci/run-dualstack-e2e-test.sh:84]: `90m` is way too long for the timeout?

**Review by ahmedwaleedmalik — DISMISSED** [review]

File: hack/ci/testdata/kubermatic_dualstack.yaml
Comment (PratikDeoghare) [reply hack/ci/testdata/kubermatic_dualstack.yaml]: I know. Was waiting for the release. Can change now that we have them.

File: hack/ci/run-dualstack-e2e-test.sh
Comment (PratikDeoghare) [reply hack/ci/run-dualstack-e2e-test.sh:84]: It is a long time. Unfortunately it is necessary because certain providers take a long time to bring nodes up e.g. Equinix.

Comment (ahmedwaleedmalik) [reply hack/ci/run-dualstack-e2e-test.sh:84]: Ah then it makes sense 👍

**Comment by PratikDeoghare** [comment]
@ahmedwaleedmalik Lets wait on retesting till https://github.com/kubermatic/kubermatic/pull/10856 merges then I will remove the hashes and we can retest.

**Comment by PratikDeoghare** [comment]
/hold there is a data race 

```
==================
WARNING: DATA RACE
Write at 0x00c000824f58 by goroutine 69:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:292 +0xae
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Previous write at 0x00c000824f58 by goroutine 68:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:292 +0xae
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 68 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c000824f48 by goroutine 69:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:293 +0x124
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Previous write at 0x00c000824f48 by goroutine 68:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:293 +0x124
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 68 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c000824f68 by goroutine 69:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:294 +0x190
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Previous write at 0x00c000824f68 by goroutine 68:
  k8c.io/kubermatic/v2/pkg/test/dualstack.addRHELSubscriptionInfo()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers.go:294 +0x190
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily.func1()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:396 +0x1b7b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 68 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1486 +0x724
  k8c.io/kubermatic/v2/pkg/test/dualstack.TestCloudClusterIPFamily()
      /home/prow/go/src/github.com/kubermatic/kubermatic/pkg/test/dualstack/cloud_providers_dualstack_test.go:348 +0x2284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================

```

File: pkg/test/dualstack/cloud_providers_dualstack_test.go
Comment (moadqassem) [inline pkg/test/dualstack/cloud_providers_dualstack_test.go:563]: What if the `teardown` length is one?

File: hack/ci/testdata/seed.yaml
Comment (moadqassem) [inline hack/ci/testdata/seed.yaml:185]: Did you tests against this release version ?

File: pkg/test/dualstack/cloud_providers_dualstack_test.go
Comment (PratikDeoghare) [reply pkg/test/dualstack/cloud_providers_dualstack_test.go:563]: In that case 
n = 1
i = 0 
n - 1 - i = 1 - 1 - 0 = 0 

all good

yes?

File: hack/ci/testdata/seed.yaml
Comment (PratikDeoghare) [reply hack/ci/testdata/seed.yaml:185]: Yes. I think so. 

The change was committed 12 days ago.

This test ran last Thursday. https://public-prow.loodse.com/view/gs/prow-dev-public-data/pr-logs/pull/kubermatic_kubermatic/10700/pre-kubermatic-dualstack-e2e-openstack-cilium/1562767405480939520

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Comment by xrstf** [comment]
Hetzner Helsinki is full.

/retest
/override pre-kubermatic-api-e2e
/override pre-kubermatic-mla-e2e
/override pre-kubermatic-opa-e2e

**Comment by xrstf** [comment]
Hetzner Helsinki is full.

/override pre-kubermatic-api-e2e
/override pre-kubermatic-mla-e2e
/override pre-kubermatic-opa-e2e

**Pull request merged**
